js/run_js_binary.bzl

```diff
@@ -0,0 +1,146 @@
+
+            Subject to `$(location)` and make variable expansion.
+
+        srcs: Additional inputs of the action.
+
+            These labels are available for `$(location)` expansion in `args` and `env`.
+
+        output_dir: Set to True if you want the output to be a directory.
+
+            Exactly one of `outs`, `output_dir` may be used.
+            If you output a directory, there can only be one output, which will be a
+            directory named the same as the target.
+
+        outs: Output files generated by the action.
+
+            These labels are available for `$(location)` expansion in `args` and `env`.
+
+        args: Command line arguments of the binary.
+
+            Subject to `$(location)` and make variable expansion.
+
+        chdir: Working directory to run the binary or test in, relative to the workspace.
+
+            By default, Bazel always runs in the workspace root.
+
+            To run in the directory containing the run_js_binary under the source tree, use
+            `chdir = package_name()` (or if you're in a macro, use `native.package_name()`).
+
+            To run in the output directory where the run_js_binary writes outputs, use
+            `chdir = "$(RULEDIR)"`
+
+            WARNING: this will affect other paths passed to the program, either as arguments or in configuration files,
+            which are workspace-relative.
+
+            You may need `../../` segments to re-relativize such paths to the new working directory.
+
+        stderr: set to capture the stderr of the binary to a file, which can later be used as an input to another target
+                subject to the same semantics as `outs`
+
+        stdout: set to capture the stdout of the binary to a file, which can later be used as an input to another target
+                subject to the same semantics as `outs`
+
+        exit_code_out: set to capture the exit code of the binary to a file, which can later be used as an input to another target
+                subject to the same semantics as `outs`. Note that setting this will force the binary to exit 0.
+                If the binary creates outputs and these are declared, they must still be created
+
+        silent_on_success: produce no output on stdout nor stderr when program exits with status code 0.
+                This makes node binaries match the expected bazel paradigm.
+
+        copy_srcs_to_bin: When True, all srcs files are copied to the output tree that are not already there.
+
+        **kwargs: Additional arguments
+    """
+    extra_srcs = []
+    if copy_srcs_to_bin:
+        copy_to_bin_name = "%s_copy_srcs_to_bin" % name
+        _copy_to_bin(
+            name = copy_to_bin_name,
+            srcs = srcs,
+            tags = kwargs.get("tags"),
+        )
+        extra_srcs = [":%s" % copy_to_bin_name]
+
+    # Automatically add common and useful make variables to the environment for js_binary targets
+    # under rules_js
+    extra_env = {
```

Comment:
I'm a bit surprised these env variables aren't provided by the upstream run_binary rule. Should they be?